disnake/guild.py

```diff
@@ -3629,6 +3633,11 @@ async def ban(
         """
         if delete_message_days is not MISSING:
             delete_message_seconds = delete_message_days * 86400
+        elif isinstance(delete_message_duration, datetime.timedelta):
+            delete_message_seconds = delete_message_duration.seconds
+        else:
+            delete_message_seconds = delete_message_duration
+
```

Comment:
we should validate for an integer too, IMO

Reply:
https://github.com/DisnakeDev/disnake/pull/659/commits/4caa07a9fc036d959b27797871139497f06a607b